src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css

```diff
@@ -18,8 +18,44 @@ for further details.
   padding the content area. So, we create a CSS var
   to make it reusable and express this intent. */
   --main-area-top-padding: 32px;
+
+  /* When the sidecar is hidden, then --main-area-padding-right must adjust.
+  Note that the "left" value, --main-area-padding-left, is declared in 
+  core-dev-dot-layout.module.css, as the left value  must be shared with the
+  navigation-header component, in order to match alignment in design specs. */
+  --main-area-padding-right: 24px;
+
+  /*
+  We want our footer to have a top border which spans
+  across both the main content and sidecar.
+  We also want our footer content to be aligned with our main content,
+  taking on the same max width and left-right padding.
+
+  To achieve this, we set the main content max width,
+  main content padding, and sidecar width as CSS variables
+  to be used in the main content, sidecar, and footer containers.
+
+  The footer itself contains an absolutely positioned :before element
+  which serves as the border, and which overflows the footer on both sides.
+  
+  The footer container ensures this border overflow does not cause scroll
+  using overflow hidden. We place an empty sidecar placeholder element
+  within this container, next to the footer, to ensure spacing is consistent.
+
+  This method requires two separate "rows" for the main content + sidecar area,
+  and for the footer area. Having to share values between them is not ideal.
+  But, it sidesteps issues in mixing overflow: hidden with position: sticky.
+  Details: https://css-tricks.com/dealing-with-overflow-and-position-sticky/
```

Comment:
As in other "weird" work, I've tried to be extra explicit about _why_ this approach was taken, with the main intent here being that with this documentation, someone in the future may be able to come up with a more elegant solution.